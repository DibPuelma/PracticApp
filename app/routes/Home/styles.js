import { StyleSheet, Dimensions } from 'react-native';

module.exports = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center', 
    justifyContent:'center',
  },
  loginColumn: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center', 
    justifyContent:'center',
    maxWidth:250
  },
  logo: {
    width: 100,
    height: 100,
    borderRadius: 100 / 2,
    backgroundColor: '#E6F4F6',
    marginBottom: 20
  },
  title: {
    color: '#888888',
    marginBottom: 30,
    fontSize: 24,
    fontWeight: 'bold'
  },
  loginButtons: {
    flexDirection: 'row',
    marginBottom: 10
  },
  fullWidthButton: {
    flex: 1,
    backgroundColor: '#3FA9F5',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5
  },
  fullWidthButtonFirst: {
    flex: 1,
    backgroundColor: '#F0F0F0',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5
  },
  fullWidthButtonFirstText: {
    fontSize: 15,
    color: '#A7A7A7',
    paddingTop: 10,
    paddingBottom: 10
  },
  fullWidthButtonText: {
    fontSize: 15,
    color: 'white',
    paddingTop: 10,
    paddingBottom: 10
  },
  facebookButton: {
    height: 32,
    width: 180,
    marginTop: 20,
    alignItems: 'center',
    borderRadius: 5

  }
});