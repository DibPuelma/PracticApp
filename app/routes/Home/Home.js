'use strict';

import React, { Component } from 'react'
import {
  StyleSheet,
  View,
  Text,
  Alert,
  Button,
  Navigator,
  TouchableHighlight,
  TouchableOpacity,
  AsyncStorage
} from 'react-native';

const FBSDK = require('react-native-fbsdk');
const { LoginButton, AccessToken, GraphRequest, GraphRequestManager } = FBSDK;

import styles from './styles';

export default class Home extends Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.loginColumn}>
          <View style={styles.logo} />
          <Text style={styles.title}>Â¡Bienvenido!</Text>
          <View style={styles.loginButtons}>
            <TouchableHighlight style={styles.fullWidthButtonFirst} underlayColor="#F7F7F7" onPress={this._goToLogin.bind(this)}>
              <Text style={styles.fullWidthButtonFirstText}>ENTRAR</Text>
            </TouchableHighlight>

            <View style={{width: 10}}></View>

            <TouchableHighlight style={styles.fullWidthButton} underlayColor="#66CCF5" onPress={this._goToRegister.bind(this)}>
              <Text style={styles.fullWidthButtonText}>REGISTRARSE</Text>
            </TouchableHighlight>
          </View>

          <View style={styles.loginButtons}>
            <LoginButton
              style={styles.facebookButton}
              publishPermissions={["publish_actions"]}
              onLoginFinished={
                (error, result) => {
                  if (error) {
                    alert("Login failed with error: " + result.error);
                  } else if (result.isCancelled) {
                    alert("Login was cancelled");
                  } else {
                    /*alert("Login was successful with permissions: " + result.grantedPermissions);*/
                    this._goToMain();
                  }
                }
              }
              onLogoutFinished={() => alert("User logged out")}/>
            </View>
        </View>
      </View>
    );
  }

  _goToLogin() {
    this.props.navigator.push({id: 'LoginPage', displayNavbar: false});
    return true;
  }

  _goToRegister() {
    this.props.navigator.push({id: 'RegisterPage', displayNavbar: false});
    return true;
  }

  _goToMain() {
    var homepage = this;
    AccessToken.getCurrentAccessToken().then(
      (data) => {
        let accessToken = data.accessToken
        alert(accessToken.toString())

        const responseInfoCallback = (error, result) => {
          if (error) {
            console.log(error)
            alert('Error fetching data: ' + error.toString());
          } else {
            console.log(result)
            //alert('Success fetching data: ' + result.toString());

            // TODO: Parse birthday to age
            var user = {name: result.first_name, lastName: result.last_name, gender: result.gender[0], age: 23, email: result.email};
            homepage.setState({user: user});

            AsyncStorage.setItem("user", JSON.stringify(user));

            homepage.props.navigator.replace({id: 'QRReader', passProps: {user: this.state.user}}); //resetTo
          }
        }

        const infoRequest = new GraphRequest(
          '/me',
          {
            accessToken: accessToken,
            parameters: {
              fields: {
                string: 'email,name,first_name,middle_name,last_name,gender,birthday'
              }
            }
          },
          responseInfoCallback
        );

        // Start the graph request.
        new GraphRequestManager().addRequest(infoRequest).start();
      }
    )

  }
}
