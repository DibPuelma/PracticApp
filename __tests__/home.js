import 'react-native';
import React from 'react';
import Home from '../app/routes/Home/Home'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Home />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
