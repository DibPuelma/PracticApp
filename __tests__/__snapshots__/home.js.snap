exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "maxWidth": 250,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "#E6F4F6",
          "borderRadius": 50,
          "height": 100,
          "marginBottom": 20,
          "width": 100,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#888888",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 30,
        }
      }>
      Â¡Bienvenido!
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#F0F0F0",
              "borderRadius": 5,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#A7A7A7",
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }>
          ENTRAR
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 10,
          }
        } />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#3FA9F5",
              "borderRadius": 5,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }>
          REGISTRARSE
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }>
      <RCTFBLoginButton
        onChange={[Function]}
        onLoginFinished={[Function]}
        onLogoutFinished={[Function]}
        publishPermissions={
          Array [
            "publish_actions",
          ]
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "height": 32,
            "marginTop": 20,
            "width": 180,
          }
        } />
    </View>
  </View>
</View>
`;
